src/io.h

```diff
@@ -75,11 +77,13 @@ typedef struct io_ops {
    io_cleanup_all_fn         cleanup_all;
    io_thread_register_fn     thread_register;
    io_max_latency_elapsed_fn max_latency_elapsed;
+   io_get_context_fn         get_context;
 } io_ops;
 
 // to sub-class io, make an io your first field;
 struct io_handle {
    const io_ops *ops;
+   int           nbytes_rw; // # of bytes read / written by IO call.
```

Comment:
Added for diagnostics. The IO call returns the # of bytes read/written.

Diagnostic messages to trace incomplete IO will report this value, as a troubleshooting aid.

Reply:
@ajhconway - I ran into some IO errors while doing other integration work and test-dev, so chased this down to understand why the IO failed.

The new field is supposed to return the # of bytes read / written on an IO call that failed.

Renamed it as you suggested.

About raising an error message: All places where `io_read()` or `io_write()` is done currently do a `platform_assert()`. I did not try to change them all to report an error message in this rework. I feel that is something that could be parceled out (if deemed important) into a follow-on issue.